backend/app/crud/evaluations/core.py

```
@@ -35,8 +35,72 @@ async def upload_dataset_to_langfuse(
     Returns:
         Tuple of (success, dataset_response, error_message)
     """
+    # Validate duplication_factor
+    if duplication_factor <= 0:
+        return False, None, "duplication_factor must be greater than 0"
+    if duplication_factor > 100:
+        return (
+            False,
+            None,
+            f"duplication_factor too large ({duplication_factor}). Maximum allowed is 100",
+        )
+
+    # Validate CSV file size (max 1MB)
+    max_size_bytes = 1_048_576  # 1MB
+    if len(csv_content) > max_size_bytes:
+        size_mb = len(csv_content) / max_size_bytes
+        return (
+            False,
+            None,
+            f"CSV file too large ({size_mb:.2f}MB). Maximum allowed is 1MB",
+        )
+
     try:
-        # Get Langfuse client
+        csv_text = csv_content.decode("utf-8")
+        csv_reader = csv.DictReader(io.StringIO(csv_text))
+
+        if not csv_reader.fieldnames:
+            return False, None, "CSV file has no headers"
```

Should not this error be raised instead of returning?